logger.js

```diff
@@ -79,7 +79,7 @@ logger.dump_and_exit = function (code) {
 
 logger.log = function (level, data) {
     if (level === 'PROTOCOL') {
-        data = data.replace(/\n/g, '\\n\n');
+        data = data.replace(/\n/g, '\\n');
```

Comment:
when you consider `.replace(/\n$/, '');` from 2 lines below, is there any point at all in doing this replace?